source/geod24/concurrency.d

```diff
@@ -1503,6 +1503,152 @@ private:
     assert(x[0] == 5);
 }
 
+/***********************************************************************
+
+    A common interface for objects that can be used in `select()`
+
+***********************************************************************/
+
+public interface Selectable
+{
+    void selectWrite (void* ptr, SelectState sel_state, int sel_id);
+    void selectRead (void* ptr, SelectState sel_state, int sel_id);
+}
+
+/***********************************************************************
+
+    An aggregate to hold neccessary information for a select operation
+
+***********************************************************************/
+
+public struct SelectEntry
+{
+    /// Reference to a Selectable object
+    Selectable selectable;
+
+    /// Pointer to the select data
+    void* select_data;
+
+    this (Selectable selectable, void* select_data)
```

Comment:
docs please, otherwise it might seem like you can partially construct it via `SelectEntry(selectable)`.